cel/library.go

```diff
@@ -479,6 +477,20 @@ func (l library) validateIPPrefix(p string, ver int64, strict bool) bool {
 	}
 }
 
+func (l library) validateURI(val string, checkAbs bool) bool {
+	uri, err := url.Parse(val)
+	if err != nil {
+		return false
+	}
+	if checkAbs && !uri.IsAbs() {
+		return false
+	}
+
+	// Parse the query string to validate it is formed and encoded properly
+	_, err = url.ParseQuery(uri.RawQuery)
```

Comment:
I understand the intent here, but this does an awful lot of work just to check the validity of the encoding. It parses out the query components and constructs a map with them, which we don't need.

It's a bit less succinct, but we can get the exact same validation while doing a lot less work:
```go
	// (Not sure if we want to disallow unescaped semicolons, but ParseQuery does.)
	if strings.Contains(uri.RawQuery, ";") {
		return false
	}
	_, err = url.QueryUnescape(uri.RawQuery)
	return err == nil
```

It'd be even better if we could just validate without parsing and decoding everything, but I argue we at least should avoid building the map if we can.

Reply:
Actually in talking to @timostamm today, we may want to discuss this deeper once his validation is done for protovalidate-ts. We can then make sure all the implementations are behaving the same way since he's following the RFC. We could then add additional conformance checks and formally define the grammar for what we validate.